core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -85,6 +86,7 @@ public class HTTPClient implements RESTClient {
 
   private HTTPClient(
       String uri,
+      HttpHost proxy,
```

Comment:
Thanks a lot of the PR!

Can we also add at least support for username/password authentication? Here's how to set it: https://stackoverflow.com/questions/6962047/apache-httpclient-4-1-proxy-authentication

Reply:
Done